iroh-gateway/src/regex.rs

```diff
@@ -0,0 +1,28 @@
+use once_cell::sync::Lazy;
+use regex::Regex;
+
+pub static SUBDOMAIN_REGEX: Lazy<Regex> = Lazy::new(|| {
+    Regex::new(r"^(?P<cid_or_domain>[A-Za-z\d\-.]+)\.(?P<scheme>ip(?:ns|fs))\.(?P<hostname>[^.]+)$")
+        .unwrap()
```

Comment:
Just to make sure there are no regressions in the future, add tests for edge cases where hostname is the same as scheme, e.g.:

`bafy.ipfs.ipfs.com` 
`bafy.ipns.ipfs.com`
`bafy.ipns.ipns.com`
`bafy.ipns.ipfs.com`

Reply:
Done